datafusion/src/physical_plan/aggregates.rs

```diff
@@ -331,17 +348,28 @@ pub fn signature(fun: &AggregateFunction) -> Signature {
         | AggregateFunction::StddevPop => {
             Signature::uniform(1, NUMERICS.to_vec(), Volatility::Immutable)
         }
+        AggregateFunction::ApproxQuantile => Signature::one_of(
+            // Accept any numeric value paired with a float64 quantile
+            NUMERICS
+                .iter()
+                .map(|t| TypeSignature::Exact(vec![t.clone(), DataType::Float64]))
```

Comment:
This is the part I am not fully sure yet. The second argument is a literal for quantile, here, to my understanding, is for checking the input schema type. I did some local test with signature and get some weird results. I will take a another look deeper and report back.

Reply:
I took another look, so for functions with literal arguments, it will be filled into an array then passed inside the function. I believe this is normal SQL behavior. For example `corr(col1, 1.0)` will take col1 and fill another array with all `1.0`s then calculate the correlation. 

For `quantile` this is not desired behavior though it should work as is. It also consumes more memory. I am wondering if there is a more elegant way to deal with function parameters. @alamb

Reply:
The long term way I hope to deal with it is via `ColumnarValue` which wraps either an `Array` or ` ScalarValue`. In this way the overhead of repeating the same `1.0` value should be entirely avoided

https://github.com/apache/arrow-datafusion/blob/master/datafusion/src/physical_plan/mod.rs#L408


```rust 
/// Represents the result from an expression
#[derive(Clone)]
pub enum ColumnarValue {
    /// Array of values
    Array(ArrayRef),
    /// A single value
    Scalar(ScalarValue),
}
```

That being said, given the current `Aggregate` interface is in terms of `Array`s rather than `ColumnarValues` there would be some more work to make the idea a reality

Reply:
Cool, leave as is good for now.